STYLE_GUIDE.md

```diff
@@ -0,0 +1,179 @@
+# Style Guide
+
+In datarockets we enforce a community [Ruby Style Guide](https://github.com/rubocop-hq/ruby-style-guide).
+
+This is a small list of differences which we have when compared with community style guide:
+
+## Bundler
+
+* A Gem's requirements should be listed only once in a Gemfile
+
+* Gems should be alphabetically sorted within groups. Also you can use a line comment as a group separator.
+
+## Style
+
+* Limit lines to 80 characters.
+
+* Prefer double-quotes unless your string literal contains " or escape characters you want to suppress.
+
+* If elements of a hash literal span more than one line we're aligning them by keys
+
+
+```ruby
+# bad
+{
+  foo: bar,
+   ba: baz
+}
+{
+  foo: bar,
+  ba:  baz
+}
+
+# good
+{
+  foo: bar,
+  ba: baz
+}
+```
+
+* There is one exception in the rule above when the hash literal is passed as method arguments.
+In this case we're just using an indentation level.
+
+```ruby
+# bad
+do_something(foo: 1,
```

Comment:
I think this looks better and more readable when you alaign params one by one like in this sample

Reply:
@TheBlackArroVV  I don't think so. I don't really like as it looks on this sample. But it looks really bad in most cases so we decided to add this rule. Let I show you:

1) Usually, we're using keyword arguments with service. + services have a really long name (Cause we're grouping it by namespaces) So as a result we can get something like 

```ruby
FreshsalesUtil::ContactsScope::FreshsalesContact.new(client: client, 
                                                     object: contact, 
                                                     fields: fields)
```

As for me, there is a big indentation here

2) other. E.g we need to call some service as one hash value (with keyword  arguments as well) 

```ruby
  FreshsalesUtil::ContactsScope::FreshsalesContact.call(client: client, 
                                                        object: FreshsalesUtil::BaseEntityFactory.call(client: client,
                                                                                                       contact: contact,
                                                                                                       type: 'lead',
                                                                                                       history: contact.history), 
                                                        fields: fields)
```

I think that something like this is better

```ruby
  FreshsalesUtil::ContactsScope::FreshsalesContact.call(client: client, 
    object: FreshsalesUtil::BaseEntityFactory.call(client: client,
      contact: contact,
      type: 'lead',
      history: contact.history), 
    fields: fields)
```

P.S. I'm agreed that my example is very weird and we can refactor and do it better. But I hope you got main ides

Reply:
But this is just a note for this rule 

https://github.com/datarockets/datarockets-style/blob/master/default.yml#L23-L25

and this is just checking that we don't make so big indentations.

Reply:
In my code, I prefer to see two styles of multiline hashes

1) Ruby style

```ruby
Module::Service(arg1: value, arg2: value, arg3: value, 
  arg4: value)
```

2)  React style

```ruby (but in this case, we have some conflict with method size metric
Module::Service(
  arg1: value, 
  arg2: value, 
  arg3: value, 
  arg4: value
)
```

Reply:
What do you think?

Reply:
so it can be 

```ruby
  FreshsalesUtil::ContactsScope::FreshsalesContact.call(client: client, fields: fields
    object: FreshsalesUtil::BaseEntityFactory.call(client: client, contact: contact,
      type: 'lead', history: contact.history))
```

or 

```ruby
  FreshsalesUtil::ContactsScope::FreshsalesContact.call(
    client: client, 
    fields: fields
    object: FreshsalesUtil::BaseEntityFactory.call(
      client: client, 
      contact: contact,
      type: 'lead', 
      history: contact.history
    )
  )
```

Reply:
Seems logic
For me is readable second variant

Reply:
the first variant looks good too if we do it right

```ruby
object = FreshsalesUtil::BaseEntityFactory.call(client: client, contact: contact,
  type: 'lead', history: contact.history)

FreshsalesUtil::ContactsScope::FreshsalesContact.call(client: client, fields: fields,
  object: object)
```

Reply:
@DDKatch yep, I think so but this is expensive from metrics length perspective (+ this metric allows to use both variants). Feel free to add a new issue if you want to choose only the second variant

Reply:
P.S. @TheBlackArroVV @DDKatch also I'll update the note to variant below. Does it work for you? 

# good
do_something(foo: 1,
  bar: 2)

# also good
do_something(
  foo: 1,
  bar: 2
)

Reply:
@roman-dubrovsky let's change the metrics then, because even you agreed that second variant is better 😉

Reply:
@DDKatch there are not metric for the second variant. We need to add own cop in this case.

Reply:
Can we go with https://github.com/datarockets/datarockets-style/pull/20/files#r288183686 after all? All of us agreed

Reply:
@AleksSenkou yep, but this rule has been already described in the style guides. I just showed another example, didn't I?

Reply:
@roman-dubrovsky makes sense 👍